doc/website/blog/2022-03-14-ent-hooks-examples.md

```diff
@@ -0,0 +1,434 @@
+	"context"
+
+	"entgo.io/ent"
+)
+
+func exampleHook() ent.Hook {
+	//use this to init your hook
+	return func(next ent.Mutator) ent.Mutator {
+		return ent.MutateFunc(func(ctx context.Context, m ent.Mutation) (ent.Value, error) {
+			// Do something before mutation.
+			v, err := next.Mutate(ctx, m)
+			if err != nil {
+				// Do something if error after mutation.
+			}
+			// Do something after mutation.
+			return v, err
+		})
+	}
+}
+```
+
+In Ent, there are two types of mutation hooks - schema hooks and runtime hooks. Schema hooks are mainly used for defining
+custom mutation logic on a specific entity type, for example, syncing entity creation to another system. Runtime hooks, on the other hand, are used
+to define more global logic for adding things like logging, metrics, tracing, etc.
+
+In this post, I will focus mainly on schemas hooks.
+
+### So why use them?
+
+In ent, we try and push the entity's logic down to the schema level, this allows us to define everything about an entity close to its creation. Entity properties could be:
+1. Valid values.
+2. Types.
+3. Allowed mutations.
+4. Privacy.
+5. Required Side effect (heavy compute for example).
+
+---
+Many times, it’s more comfortable to define that in the RPC level, a controller provides a context and all the mutations
+and validation are done before changing the value.
+Yet, as the system gets bigger and more developers are working on it making sure that these validations and calculations are done
+whenever the entity is changing becomes harder and requires more pre-known knowledge about each entity to work with and
+can cause unintentional bugs or inconsistencies in the data that are only discovered in runtime.
+The way ent, based on the experience of bigger companies like Facebook, decided to handle it is to keep this information
+as close as possible to the entity, therefore hooks are just that, a piece of code that runs before and after an entity 
+is changing and defines that additional behavior.
+
+## Let us do some examples
+
+For this examples we use a simple schema of users and dogs.
+Each user can have multiple pets and each pet has an owner.
+We have also added a cache entity that will be used later.
+as can be seen here:
+```go
+// User holds the schema definition for the User entity.
+type User struct {
+	ent.Schema
+}
+
+// Fields of the User.
+func (User) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name").
+			NotEmpty(),
+		field.String("connection_string").
+			NotEmpty(),
+		field.String("password").
+			Sensitive().
+			NotEmpty(),
+	}
+}
+
+// Edges of the User.
+func (User) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("pets", Dog.Type),
+		edge.To("cache", Cache.Type).Unique(),
+	}
+}
+```
+```go
+// Dog holds the schema definition for the Dog entity.
+type Dog struct {
+	ent.Schema
+}
+
+// Fields of the Dog.
+func (Dog) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name").
+			NotEmpty(),
+		field.Int("owner_id").
```

Comment:
Why do you add this field here (and `Field("owner_id")` on the edge)? It adds more code but no value for the posts topic

Edit: I now see why you need them. Add a short sentence stating, that the field is needed later.